pkg/syncer/syncers/common.go

```diff
@@ -182,7 +183,7 @@ func diff(logger klog.Logger, srcResourceList *unstructured.UnstructuredList, de
 		if ok {
 			srcResource.SetResourceVersion(destResource.GetResourceVersion())
 			srcResource.SetUID(destResource.GetUID())
-			srcResource.SetManagedFields(destResource.GetManagedFields())
+			srcResource.SetManagedFields([]metav1.ManagedFieldsEntry{})
```

Comment:
If the given value were `nil` instead of an empty slice then this would hit the even more satisfying special case at the start of SetManagedFields:

```
func (u *Unstructured) SetManagedFields(managedFields []metav1.ManagedFieldsEntry) {
	if managedFields == nil {
		RemoveNestedField(u.Object, "metadata", "managedFields")
		return
	}
...
```

Reply:
But an empty slice is equivalent on the wire.  See https://github.com/kubernetes/apimachinery/blob/v0.24.3/pkg/apis/meta/v1/types.go#L275

Reply:
Thanks. Using `nil` might be better. I tried it at local and confirmed that managedFields are also not overwritten by Syncer.

Reply:
I think it would only be better in the sense of making its intended effect more obvious. But, as I noted above, because of the way that objects are serialized and deserialized --- both to/from JSON and protobuf --- `nil` is equivalent to empty slice.